CondFormats/HGCalObjects/interface/HGCalMappingCellIndexerTrigger.h

```diff
@@ -0,0 +1,177 @@
+#ifndef CondFormats_HGCalObjects_interface_HGCalMappingCellIndexerTrigger_h
+#define CondFormats_HGCalObjects_interface_HGCalMappingCellIndexerTrigger_h
+
+#include <iostream>
+#include <cstdint>
+#include <vector>
+#include <numeric>
+#include "CondFormats/Serialization/interface/Serializable.h"
+#include "CondFormats/HGCalObjects/interface/HGCalDenseIndexerBase.h"
+
+/**
+   @short utility class to assign dense readout trigger cell indexing
+ */
+class HGCalMappingCellIndexerTrigger {
+public:
+  // typedef HGCalDenseIndexerBase WaferDenseIndexerBase;
+
+  HGCalMappingCellIndexerTrigger() = default;
+
+  /**
+     adds to map of type codes (= module types) to handle and updatest the max. number of eRx
+   */
+  void processNewCell(std::string typecode, uint16_t ROC, uint16_t trLink, uint16_t trCell) {
+    // Skip if trLink, trCell not connected
+    if (trLink == uint16_t(-1) || trCell == uint16_t(-1))
+      return;
+    //assign index to this typecode and resize the max vectors
+    if (typeCodeIndexer_.count(typecode) == 0) {
+      typeCodeIndexer_[typecode] = typeCodeIndexer_.size();
+      maxROC_.resize(typeCodeIndexer_.size(), 0);
+      maxTrLink_.resize(typeCodeIndexer_.size(), 0);
+      maxTCPerLink_.resize(typeCodeIndexer_.size(), 0);
+    }
+
+    size_t idx = typeCodeIndexer_[typecode];
+
+    /*High density modules have links {0, 2} and low {0, 1, 2, 3} so to make it work I need to divide by 2*/
+    if (typecode[1] == 'H')
+      trLink /= 2;
+    /*SiPM tiles have trCells indexed from 1 instead from 0*/
+    if (typecode[0] == 'T')
+      trCell--;
+    maxROC_[idx] = std::max(maxROC_[idx], static_cast<uint16_t>(ROC + 1));
+    maxTrLink_[idx] = std::max(maxTrLink_[idx], static_cast<uint16_t>(trLink + 1));
+    maxTCPerLink_[idx] = std::max(maxTCPerLink_[idx], static_cast<uint16_t>(trCell + 1));
+  }
+
+  /**
+     @short process the current list of type codes handled and updates the dense indexers
+  */
+  void update() {
+    uint32_t n = typeCodeIndexer_.size();
+    offsets_ = std::vector<uint32_t>(n, 0);
+    di_ = std::vector<HGCalDenseIndexerBase>(n, HGCalDenseIndexerBase(3)); /* The indices are {ROC, trLink, TC}*/
+    for (uint32_t idx = 0; idx < n; idx++) {
+      uint16_t maxROCs = maxROC_[idx];
+      uint16_t maxLinks = maxTrLink_[idx];
+      uint16_t maxTCPerLink = maxTCPerLink_[idx];
+      di_[idx].updateRanges({{maxROCs, maxLinks, maxTCPerLink}});
+      if (idx < n - 1)
+        offsets_[idx + 1] = di_[idx].maxIndex();
+    }
+
+    //accumulate the offsets in the array
+    std::partial_sum(offsets_.begin(), offsets_.end(), offsets_.begin());
```

Comment:
Since you are already looping over the values, isn't it simpler to keep a counter and update it inside the loop, instead of storing the indices and then doing a second loop ?